app/src/lib/stores/app-store.ts

```diff
@@ -968,7 +966,7 @@ export class AppStore extends TypedBaseStore<IAppState> {
       isUpdateShowcaseVisible: this.isUpdateShowcaseVisible,
       currentBanner: this.currentBanner,
       askToMoveToApplicationsFolderSetting:
-        this.askToMoveToApplicationsFolderSetting,
+        this.localStorageManager.get('image-diff-type'),
```

Comment:
This is highlighting [my other comment](https://github.com/desktop/desktop/pull/17386/files#r1324938857) that the casting of the type is bad. :D Because the `image-diff-type` is a object.. but this doesn't complain even tho `askToMoveToApplicationsFolderSetting` is a boolean.

Reply:
In this case `image-diff-type` is a number, right? Hence why it works at runtime, because it's able to convert it to a boolean.

However, `get` is inferring `T` from the _required_ return type, which is a `boolean` thanks to `askToMoveToApplicationsFolderSetting`, instead of inferring it from the type of the value for the given `image-diff-type` key, which I think can't be done 😞

Reply:
This idea made the trick:
1. First remove the `Record…` typing from the defaults object:
```typescript
const LocalStorageDefaults = { ... } as const
```
2. Make `get` generic parameter to infer the key type (which should be the string itself) and then use that to parameterize the return type:
```typescript
  public get<
    V extends LocalStorageKey,
    T extends typeof LocalStorageDefaults[V]
  >(key: V): T {
```

The only issue I see with this is that for nullable values, you need to be explicit about the type in the defaults:
```typescript
  'version-and-users-of-last-thank-you': null as object | null,
```

This is what this line looks like now:
![image](https://github.com/desktop/desktop/assets/1083228/a01cc807-3256-4096-b377-c9d6eeff9938)
![image](https://github.com/desktop/desktop/assets/1083228/1e8704ab-df49-4a64-b545-ad86cb3a40b2)